interoperator/pkg/controller/sfplan/sfplan_controller.go

```diff
@@ -76,18 +77,29 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-func restartOnWatchUpdate(mgr manager.Manager, initWatches, stop <-chan struct{}) {
+func reconfigureWatches(mgr manager.Manager, reconfigureWatch, stop <-chan struct{}) {
 	for {
 		select {
-		case <-initWatches:
-			toUpdate, err := watches.InitWatchConfig(mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper())
+		case <-reconfigureWatch:
+			drainTimeout := time.After(constants.ChannelDrainTimeout)
```

Comment:
This is done so that during watch refreshes, lets say there are total 50 sfplans, reconcile gets called 50 times and will result in recomputing watch lists from sources templates 50 times. So a logic is added that will drain all the reconcile requests happening within .5 sec and recompute watch list only once. This was done just for performance improvement. The entire Drain loop and this timeout is not functionally required.

Reply:
No. I understood the requirement. My question was about the implementation/solution. I was asking if instead of draining the watch channel for a time period, if we can use a timestamp and and ignore the reconciliation if we are still within the time period. Could reduce additional goroutine/parallel flows and keep the logic simple.